tron/serialize/runstate/statemanager.py

```diff
@@ -160,15 +174,16 @@ def restore(self, job_names, skip_validation=False):
         return state
 
     def _restore_runs_for_job(self, job_name, job_state):
+        """Restore the state for the runs of each job"""
         run_nums = job_state["run_nums"]
         runs = []
+        keys_ids_list = []
+        # with self._lock:
         for run_num in run_nums:
             key = jobrun.get_job_run_id(job_name, run_num)
-            run_state = list(self._restore_dicts(runstate.JOB_RUN_STATE, [key]).values())
-            if not run_state:
-                log.error(f"Failed to restore {key}, no state found for it")
-            else:
-                runs.append(run_state[0])
+            keys_ids_list.append(key)
+        run_state = list(self._restore_dicts(runstate.JOB_RUN_STATE, keys_ids_list).values())
+        runs.extend(run_state)
```

Comment:
question: could we return `run_state` here?

Reply:
Nice good catch, I think I overlooked this when trying to have this new code behave the same way this func was written before